binr/rahash2/rahash2.c

```diff
@@ -288,10 +288,80 @@ int is_power_of_two(const ut64 x) {
 	return x && !(x & (x - 1));
 }
 
+int encrypt_or_decrypt (const char *algo, bool to_encrypt, const char *hashstr, int hashstr_len) {
+	bool no_key_mode = !strcmp ("base64", algo) || !strcmp ("base91", algo); //TODO: generalise this for all non key encoding/decoding.
```

Comment:
in libr/crypt/crypto.c. you must just check if `crypto->h->set_key != NULL` but having a getter in `bool r_crypto_need_key(RCrypto *crypto) { return crypto->h->set_key != NULL; }` can be nice too.

feel free to update the vapi files for the bindings